pkg/daemon/update.go

```diff
@@ -174,10 +181,16 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) error {
 		return err
 	}
 
+	// TODO(runcom):
+	// roll back updateFiles on error
```

Comment:
That's...going to be nontrivial.

Reply:
I may be complitely stupid, but isn't this just a matter of inverting the argument to `updateFiles`?

```
	// update files on disk that need updating
	if err := dn.updateFiles(newConfig, oldConfig); err != nil {
		return err
	}
```

I'm sure I'm missing something obvious as I've not yet thought about this yet that much

Reply:
ooo I have the solution actually https://github.com/openshift/machine-config-operator/pull/401

I'm modify the atomic write PR to split the writing into a real transation and only commit at the very end! I'll try that

Reply:
Yeah, that could work.  I think where it gets tricky is trying to handle an error when we're in the *middle* of either writing files or doing a revert.  Then we're in an "indeterminate middle state", much like what happens in the middle of a `yum/apt update`.

But anyways for now we can probably ignore that and yeah, just delete any files added by the new config and restore all the files from the old one.

Reply:
I'm ignorant on `yum/apt update` - how are those handling such a failure in the middle of an update?